client-spark/spark2/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -177,19 +177,27 @@ public RssShuffleManager(SparkConf sparkConf, boolean isDriver) {
     sparkConf.set("spark.shuffle.service.enabled", "false");
     LOG.info("Disable external shuffle service in RssShuffleManager.");
     if (!sparkConf.getBoolean(RssSparkConfig.RSS_TEST_FLAG.key(), false)) {
-      // for non-driver executor, start a thread for sending shuffle data to shuffle server
-      LOG.info("RSS data send thread is starting");
-      eventLoop.start();
-      int poolSize = sparkConf.get(RssSparkConfig.RSS_CLIENT_SEND_THREAD_POOL_SIZE);
-      int keepAliveTime = sparkConf.get(RssSparkConfig.RSS_CLIENT_SEND_THREAD_POOL_KEEPALIVE);
-      threadPoolExecutor = new ThreadPoolExecutor(poolSize, poolSize * 2, keepAliveTime, TimeUnit.SECONDS,
-          Queues.newLinkedBlockingQueue(Integer.MAX_VALUE),
-          ThreadUtils.getThreadFactory("SendData-%d"));
-
       if (isDriver) {
         heartBeatScheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
             ThreadUtils.getThreadFactory("rss-heartbeat-%d"));
       }
+
+      if (!isDriver || String.valueOf(true).equals(System.getProperty("TEST"))) {
```

Comment:
Could we use a config option? We have a config option named `RSS_TEST_FLAG`, Should we keep consistent with it?

Reply:
Agree.